### DISCUSSION TITLE
Build for rp2040 without the sdk?

### DISCUSSION THREAD

**Discussion by jakerr** [hathach/tinyusb#2894] [Q&A] [answered]
Would it be possible to build a static library of tinyusb that I could call into from an rp2040 chip but without depending on the pico sdk?

I'd like to build a rust wrapper for it that I can include in an embassy-rs project and initialize and then manually call the interrupt tasks.

I've made it as far as linking in the existing 2040 library into my embassy project and I can call into it but when I call init I get failed assertions from within pico_sdk code and feel like I'm working against the grain by both having the embassy runtime and pico sdk runtime trying to manage interrupts.

Rather than trying to give pico_sdk everything it needs via link scripting gymnastics perhaps I can make a new target that is just more bare bones.

Is there anything that would be prohibitively 
difficult to implement without the sdk?

**Comment by JustAnother1** [comment, marked as answer]
I don't know if it helps. But I have tinyusb working on a RP2040 without the sdk. Code is here: https://github.com/JustAnother1/nomagic_probe/tree/main/src/tinyusb

**Reply by jakerr** [reply]
Hi! Thanks so much for taking the time to share that. I'll have some time to look at it in more detail later this week. I'll be sure to post an update on how it goes.

**Reply by jakerr** [reply]
Thanks again!
I'll mark this as resolved as it's clear that the answer to my question "would it be possible..." is yes.

**Comment by piersfinlayson** [comment]
I found myself wanting to build tinyusb for the RP2350 without the SDK, so created

https://github.com/piersfinlayson/pico-sdkless

It contains a complete working bare-metal example of using tinyusb to implement a CDC device on an RP2350 without the Pico SDK. 

No edits are required to the tinyusb source to make this work  - it essentially provides the Pico SDK header file names tinyusb expects, and the information needed.
